apps/typegpu-docs/src/content/examples/simple/vaporrave/floor.ts

```diff
@@ -0,0 +1,46 @@
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+
+import * as c from './constants.ts';
+
+export const grid = tgpu.fn([d.vec2f, d.f32], d.vec3f)((uv, time) => {
+  // time is really an angle, but we are fine as long as it keeps increasing
+  const uvNormalized = std.fract(
+    d.vec2f(uv.x, uv.y + time).div(c.GRID_SEP),
+  );
+
+  // x^4 + y^4 = 0.5^4
+  const diff4 = std.pow(
+    d.vec2f(0.5, 0.5).sub(uvNormalized),
+    d.vec2f(4, 4),
+  );
+  const sdf = std.pow(diff4.x + diff4.y, 0.25) - 0.5; // -radius
+
+  return std.mix(
+    c.gridInnerColor,
+    c.gridColor,
+    std.exp(c.GRID_TIGHTNESS * sdf), // fading color
+  );
+});
+
+export const circles = tgpu.fn([d.vec2f, d.f32], d.vec3f)((uv, angle) => {
+  const rotMatY = d.mat4x4f.rotationY(angle);
+  const uvRotated = d.vec4f(uv.x, 1.0, uv.y, 1)
+    .add(d.vec4f(0, 0, -c.sphereCenter.z, 0))
+    .mul(rotMatY);
```

Comment:
I'm being very nitpicky, but since we don't need affine transformations (in 4 dimensions), and we only care about rotation in a single plane, we can rotate in 2d around an imaginary Z axis.

```suggestion
  const uvRotated = rotateXY(angle).mul(d.vec2f(uv.x, uv.y - c.sphereCenter.z));
```

I wrote a helpful function for doing just that for the Caustics example, and you paste it above `circles`:
```ts
/**
 * Returns a transformation matrix that represents an `angle` rotation
 * in the XY plane (around the imaginary Z axis)
 */
const rotateXY = tgpu.fn([d.f32], d.mat2x2f)((angle) =>
  d.mat2x2f(
    /* right */ d.vec2f(std.cos(angle), std.sin(angle)),
    /* up    */ d.vec2f(-std.sin(angle), std.cos(angle)),
  )
);
```